pkg/k8s/watchers/pod.go

```diff
@@ -214,6 +214,10 @@ func (k *K8sWatcher) addK8sPodV1(pod *slim_corev1.Pod) error {
 		return k.deleteK8sPodV1(pod)
 	}
 
+	if port := k8sUtils.GetNoTrackPort(pod.ObjectMeta.Annotations); port != "" {
```

Comment:
Yea this is where I'm a bit hesitant, if we are to keep these rules specific to nodelocaldns or something that WILL have LRP then I guess it makes sense to let `redirectPolicyManager` handle it, but if we are trying to make this generic then maybe not? I discussed this a bit with @aditighag , maybe she can provide some insights here :)

Reply:
The annotation is currently defined at a pod level so I don't see a reason for tying this to the redirect policy manager. Since the annotation could potentially be used for generic cases where NO_TRACK iptable rules need to be installed, it might be better to keep it decoupled from the LRP code?

Reply:
My take is that pks/k8s/watchers main responsibility is to retrieve objects from the kube api-server, and then notify consumers about them.

If we decide that the NOTRACK rule should be applicable only to pods participating in LRP, then it makes sense that the LRP manager should take care of invoking iptables. 

So far I haven't seen any ask to add the NOTRACK rules for a given pod. Therefore, I'd be in favor of not making this generic.

Reply:
I think we are confusing two different things here (probably caused by the current set of changes in this code path). 

The way I see it, there are two aspects to the iptable rules management - 1) monitor the relevant annotation 2) install/delete rules. For 2), I agree that this code needs to reside outside of the watcher code, which is currently the case. I think the `IptablesManager` interface seems to be the correct place to add these new methods since it also handles adding iptable rules for proxy, etc. For 1), we need to monitor for any changes in the pod annotations. This can happen in the `updateK8sPodV1` [1] code path, similar to how other annotations like proxy, bandwidthManager, etc are handled [2]. @Weil0ng  Why are we not handling the NOTRACK annotations in the pod update callback in a similar fashion? How is the case handled where pod annotations are added after the pod is deployed? 

[1] https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/pod.go#L275
[2] https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/pod.go#L329

Reply:
The expectation is that, for node-local-dns, we would deploy the daemonset with such annotation, I'm not expecting this to be a manual process for users to add the annotation. But again this assumption is specific to nodelocaldns, I'm unsure whether we should consider more generic cases...

Reply:
For reference, in `updateK8sPodV1()` and `RunMetadataResolver()` we handle the existing annotations configuration so I would at minimum suggest following the existing code paths there. If we want to reconsider how much of this logic is in `pkg/k8s/watchers`, I think that's reasonable but let's at least put all of the annotations handling together so it is consistently handled and we can refactor it all together later on if we decide there's a better structure/location for handling it.

Reply:
Updated code using suggested code paths.